### PR TITLE
Fixed Name Extraction Issues

### PR DESCRIPTION
**Pull request by chakrabortyr**
## Description
The method getFullName did not properly delimit on whitespace names prior to this change.

## Motivation and Context
Bug fix.

## Tests performed
Manual inputs to getFullName function.

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project as found in the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by plessbd** [comment]
The license should be changed in a different pull request.

Is there a PR in any of the xdmod repos to upgrade to the new version?

**Review by plessbd — CHANGES_REQUESTED** [review]
Move license change to a different PR, Add more detailed testing notes, Open PR's in related repos with updated module version

File: lib/Auth/Source/Globus.php [hunk 1]
```diff
@@ -92,20 +92,25 @@ protected function samlize($array, $prefix = '') {
     protected function getFullName($name) {
```
Comment (plessbd) [inline lib/Auth/Source/Globus.php]: this seems overly complicated compared to something like this:
```php
function getFullName($name){
    $trimmedName = trim($name);
    $lastName = 'UNKNOWN';
    $firstName = 'UNKNOWN';
    if(strpos($trimmedName, ',') !== false ){
        list($lastName, $firstName) = explode(',', $trimmedName, 2);
    }
    else if(strpos($trimmedName, ' ') !== false ){
        list($firstName, $lastName) = explode(' ', $trimmedName, 2);
    }
    return array(
        'first_name' => trim($firstName),
        'last_name' => trim($lastName)
    );
}
```

old:
```
Array
(
    [first_name] => UNKNOWN
    [last_name] => UNKNOWN
)
Array
(
    [first_name] => UNKNOWN
    [last_name] => UNKNOWN
)
Array
(
    [first_name] => 
    [last_name] => 
)
Array
(
    [first_name] => UNKNOWN
    [last_name] => UNKNOWN
)
Array
(
    [first_name] => First
    [last_name] => Last
)
Array
(
    [first_name] => First
    [last_name] => Middle Last
)
Array
(
    [first_name] => First
    [last_name] => von Last
)
Array
(
    [first_name] => First
    [last_name] => M. Last
)
Array
(
    [first_name] => First
    [last_name] => M. von Last
)
Array
(
    [first_name] => First
    [last_name] => Last
)
Array
(
    [first_name] => First Middle
    [last_name] => Last
)
Array
(
    [first_name] => First
    [last_name] => von Last
)
Array
(
    [first_name] => First M.
    [last_name] => Last
)
Array
(
    [first_name] => First M.
    [last_name] => von Last
)
```
new:
```
Array
(
    [first_name] => UNKNOWN
    [last_name] => UNKNOWN
)
Array
(
    [first_name] => UNKNOWN
    [last_name] => UNKNOWN
)
Array
(
    [first_name] => 
    [last_name] => 
)
Array
(
    [first_name] => UNKNOWN
    [last_name] => UNKNOWN
)
Array
(
    [first_name] => First
    [last_name] => Last
)
Array
(
    [first_name] => First
    [last_name] => Middle Last
)
Array
(
    [first_name] => First
    [last_name] => von Last
)
Array
(
    [first_name] => First
    [last_name] => M. Last
)
Array
(
    [first_name] => First
    [last_name] => M. von Last
)
Array
(
    [first_name] => First
    [last_name] => Last
)
Array
(
    [first_name] => First Middle
    [last_name] => Last
)
Array
(
    [first_name] => First
    [last_name] => von Last
)
Array
(
    [first_name] => First M.
    [last_name] => Last
)
Array
(
    [first_name] => First M.
    [last_name] => von Last
)
```

File: lib/Auth/Source/Globus.php [hunk 2]
```diff
@@ -92,20 +92,25 @@ protected function samlize($array, $prefix = '') {
...
         $trimmedName = trim($name);
 
-        if ($trimmedName && (strpos($trimmedName, ',') !== null || strpos($trimmedName, '\s') !== null)) {
-            $delimiter = strpos($name, ',') !== null ? ',' : '\s';
+        if ($trimmedName && (strpos($trimmedName, ',') !== false || strpos($trimmedName, ' ') !== false)) {
+            $delimiter = strpos($name, ',') !== false ? ',' : ' ';
```
Comment (plessbd) [inline lib/Auth/Source/Globus.php]: this should use `$trimmedName` instead of `$name`

**Review by plessbd — DISMISSED** [review]

File: lib/Auth/Source/Globus.php [hunk 3]
```diff
@@ -118,6 +123,24 @@ protected function getFullName($name) {
             'first_name' => 'UNKNOWN',
             'last_name' => 'UNKNOWN'
         );
+
+        /*
```
Comment (plessbd) [inline lib/Auth/Source/Globus.php]: this doesnt need to be in the code, this should be a unit test, but could just be a comment on the PR

**Comment by plessbd** [comment]
fix your errors

**Review by jpwhite4 — APPROVED** [review]
I begrudgingly approve

**Pull request merged**